torchvision/io/video.py

```diff
@@ -137,12 +137,14 @@ def _read_from_stream(container, start_offset, end_offset, pts_unit, stream, str
         pass
     # ensure that the results are sorted wrt the pts
     result = [frames[i] for i in sorted(frames) if start_offset <= frames[i].pts <= end_offset]
-    if start_offset > 0 and start_offset not in frames:
+    if len(frames) > 0 and start_offset > 0 and start_offset not in frames:
```

Comment:
nit: this check in here is not necessary. Only the second one `if len(preceding_framses) > 0)` is actually necessary I believe.